File: Platform/codesignal/Intro/12.SortByHeight/solution.go
package main

import (
	"fmt"
	"sort"
)

func solution(a []int) []int {
	p := make([]int, 0)
	for _, v := range a {
		if v != -1 {
			p = append(p, v)
		}
	}

	sort.Ints(p)
	index := 0
	for i, v := range a {
		if v != -1 {
			a[i] = p[index]
			index++
		}
	}

	return a
}

func main() {
	arr := []int{-1, 150, 190, 170, -1, -1, 160, 180}
	fmt.Println(solution(arr))
}

File: Platform/codesignal/Intro/12.SortByHeight/solution.js
function solution(a) {
  let people = a.filter( x=> x !== -1).sort((a,b) => a-b);
  a.forEach((x,i) => {
    if(x !== -1) {
      a[i] = people.shift();
    }
  })
  return a;
}

function solution2(a) {
  let arr = a.filter((x) => x != -1).sort((a, b) => a - b);
  let cnt = 0;
  return a.map((x) => (x === -1 ? -1 : arr[cnt++]));
}

console.log(solution([-1, 150, 190, 170, -1, -1, 160, 180]));